lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -52,13 +52,24 @@ def get_collection_versions(self, requirement):
             if isinstance(requirement.src, GalaxyAPI)
             else self._apis
         )
-        return set(
-            (version, api)
-            for api in api_lookup_order
-            for version in api.get_collection_versions(
-                requirement.namespace, requirement.name,
-            )
-        )
+        version_information = set()
+        for api in api_lookup_order:
+            # TODO: Should this try/except logic be in api.py?
```

Comment:
Probably not: api.GalaxyAPI encapsulates a connection to one server/endpoint but there's really no abstraction for querying things across servers + this one also attempts to treat access to the local artifacts the same way w/o any network I/O, like a facade.